cmd/kube-controller-manager/app/core.go

```diff
@@ -349,20 +347,16 @@ func startGarbageCollectorController(ctx ControllerContext) (bool, error) {
 	}
 
 	gcClientset := ctx.ClientBuilder.ClientOrDie("generic-garbage-collector")
-
-	// Use a discovery client capable of being refreshed.
 	discoveryClient := cacheddiscovery.NewMemCacheClient(gcClientset.Discovery())
```

Comment:
You need to call invalidate in the GCE sync loop because on this because it is a different instance the one used for the RESTMapper.

Reply:
I am confused, seems this discoveryClient  is only used once by `	deletableResources := garbagecollector.GetDeletableResources(discoveryClient)`.  It is not used in sync loop, so keep it as it is now seems ok. What do I miss?

Reply:
My mistake